src/test/java/org/biouno/unochoice/issue72105/TestJobDslWithScriptlerScriptWithParameters.java

```diff
@@ -0,0 +1,141 @@
+import org.biouno.unochoice.AbstractScriptableParameter;
+import org.biouno.unochoice.CascadeChoiceParameter;
+import org.biouno.unochoice.model.ScriptlerScript;
+import org.jenkinsci.plugins.scriptler.ScriptlerHelper;
+import org.jenkinsci.plugins.scriptler.ScriptlerManagement;
+import org.jenkinsci.plugins.scriptler.config.Parameter;
+import org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval;
+import org.jenkinsci.plugins.scriptsecurity.scripts.languages.GroovyLanguage;
+import org.jenkinsci.plugins.workflow.flow.FlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * In JENKINS-72105 the plug-in did not persist the user option to use
+ * parameters in their scriptler scripts, when the script was created
+ * by a Job DSL build.
+ */
+@Issue("JENKINS-72105")
+public class TestJobDslWithScriptlerScriptWithParameters {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private ScriptlerManagement scriptler;
+    private File scriptFile = null;
+
+    private static final String GROOVY_SCRIPT = "return [P1, P2]";
+    private static final String PARAMETER_WITH_SCRIPTLER_AND_PARAMETERS = "P3";
+
+    /**
+     * Create the script in Scriptler.
+     *
+     * @throws Exception I/O?
+     */
+    @Before
+    public void setUp() throws Exception {
+        ScriptApproval.get().preapprove(GROOVY_SCRIPT, new GroovyLanguage());
+        // scriptler setup
+        scriptler = j.getInstance().getExtensionList(ScriptlerManagement.class).get(0);
+        ScriptlerHelper scriptlerHelper = new ScriptlerHelper(scriptler);
+        scriptFile = Files.createTempFile("uno-choice", "Cascade.groovy").toFile();
+        FileUtils.writeStringToFile(scriptFile, GROOVY_SCRIPT, Charset.defaultCharset(), false);
+        FileItem fi = new FileParameterValue.FileItemImpl(scriptFile);
+        scriptlerHelper.saveScript(fi, true, "Cascade.groovy");
+
+        scriptler.getConfiguration()
+                .getScriptById("Cascade.groovy")
+                .setParameters(Arrays.asList(
+                        new Parameter("P1", ""),
+                        new Parameter("P2", "")));
+    }
+
+    @Test
+    @LocalData("test")
+    public void test() throws Exception {
+        assertNull(j.jenkins.getItem("scriptler-test"));
+
+        final FreeStyleProject job = (FreeStyleProject) j.jenkins.getItem("job");
+        j.buildAndAssertStatus(Result.UNSTABLE, job);
```

Comment:
Not clear to my why UNSTABLE, but the job runs fine and produces the `scriptler-test` parameter (same locally :shrug:)

Reply:
Passing now after fixing master and rebasing here.